client/src/client.ts

```diff
@@ -133,23 +147,23 @@ export class TokenServiceClient {
     return `${root}?${keyValues}`;
   }
 
-  private fetch(method: string, path: string, options?: { body?: any, readWriteToken?: string }) {
+  private fetchFromServiceUrl(method: string, path: string, options?: { body?: any, readWriteToken?: string }) {
     return new Promise<any>((resolve, reject) => {
       const url = `${this.serviceUrl}${path}`;
       const requestOptions: RequestInit = {
         method,
-        headers: new Headers({
+        headers: {
           "Content-Type": "application/json"
-        })
+        }
       };
       const authToken = options?.readWriteToken || this.jwt;
       if (authToken) {
-        (requestOptions.headers as Headers).set("Authorization", `Bearer ${authToken}`);
+        (requestOptions.headers as any)["Authorization"] = `Bearer ${authToken}`;
       }
       if (options?.body) {
         requestOptions.body = JSON.stringify(options.body);
       }
-      return fetch(url, requestOptions)
+      return this.fetch(url, requestOptions)
```

Comment:
See note about throwing an exception in the constructor as `this.fetch` could be undefined here if not found in the options (node) or in the window object (browser).